src/main/org/codehaus/groovy/transform/trait/TraitComposer.java

```diff
@@ -293,7 +293,8 @@ private static void createForwarderMethod(
 
         ClassNode[] exceptionNodes = correctToGenericsSpecRecurse(genericsSpec, copyExceptions(helperMethod.getExceptions()));
         ClassNode fixedReturnType = correctToGenericsSpecRecurse(genericsSpec, helperMethod.getReturnType());
-        Expression forwardExpression = genericsSpec.isEmpty()?mce:new CastExpression(fixedReturnType,mce);
+        boolean noCastRequired = genericsSpec.isEmpty() || fixedReturnType.getName().equals(ClassHelper.VOID_TYPE.getName());
```

Comment:
Just curious if you saw problems using just `fixedReturnType.equals(ClassHelper.VOID_TYPE)` that neccessitated comparing by name?  Most places that I see that compare types don't see to use the getName so am just curious.